modules/autotagging-commons/common/src/main/java/org/opensearch/rule/service/IndexStoredRulePersistenceService.java

```diff
@@ -52,9 +54,24 @@
  * @opensearch.experimental
  */
 public class IndexStoredRulePersistenceService implements RulePersistenceService {
+    // Default value for max rules count
+    public static final int MAX_ALLOWED_RULE_COUNT = 10000;
+
+    // max wlm rules setting name
+    public static final String MAX_RULES_COUNT_SETTING_NAME = "wlm.autotagging.max_rules";
+
+    // max wlm rules setting
+    public static final Setting<Integer> MAX_WLM_RULES_SETTING = Setting.intSetting(
+        MAX_RULES_COUNT_SETTING_NAME,
+        MAX_ALLOWED_RULE_COUNT,
```

Comment:
`10000` seems bit high to me. We should also think if that is user friendly to consume and manage. IMO, even 50-100 should be a reasonable limit, we can keep 250/500 to be on the safe side

Reply:
I agree w/ @jainankitk . Even programmatically, maintaining a 10K rules would be a nightmare. Anything under 1K is fine.